### ISSUE TITLE
FIRO SOLO MINING ERROR

### ISSUE THREAD

**Issue by sergeyzh71** [firoorg/ccminer#68]
Helo!
I used ccminer for firo solo mining/ Version ccminer 1.3.2; version firo wallet - v0.14.1.2.
Params:
 ccminer.exe -a mtp -o http://192.168.110.20:8382 -u zcoin -p 111111 --coinbase-addr xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx --no-stratum --no-getwork -i 22  
Have a next error:
2021-01-06 23:25:16] mtp block 334822 , target 593.481
[2021-01-06 23:27:22] GPU #2: GeForce GTX 1660 Ti: 1921.6 Kh/s nonce b896aaaa
[2021-01-06 23:27:25] GPU #1: GeForce RTX 2070: 2831.2 Kh/s nonce 6a375555
[2021-01-06 23:27:25] GPU #0: GeForce RTX 2070: 2789.7 Kh/s nonce 14e20000
[2021-01-06 23:29:24] GPU #2: GeForce GTX 1660 Ti: 1921.7 Kh/s nonce c6a6aaaa
[2021-01-06 23:29:30] GPU #1: GeForce RTX 2070: 2827.5 Kh/s nonce 7f4f5555
[2021-01-06 23:29:32] GPU #0: GeForce RTX 2070: 2789.7 Kh/s nonce 29fa0000
[2021-01-06 23:31:27] GPU #2: GeForce GTX 1660 Ti: 1921.6 Kh/s nonce d4b6aaaa
[2021-01-06 23:31:35] GPU #1: GeForce RTX 2070: 2826.1 Kh/s nonce 94675555
[2021-01-06 23:31:39] GPU #0: GeForce RTX 2070: 2789.2 Kh/s nonce 3f120000
[2021-01-06 23:32:34] GPU #2: GeForce GTX 1660 Ti, 1913.01 kH/s
[2021-01-06 23:32:34] accepted: 0/1 (diff 4077.514), 1913.01 kH/s booooo
[2021-01-06 23:32:34] reject reason: rejected
[2021-01-06 23:32:34] Current block is 334822
[2021-01-06 23:33:40] GPU #1: GeForce RTX 2070: 2826.3 Kh/s nonce a97f5555

Errors in wallet log:
021-01-06 08:11:09 CheckBlock() nHeight=2147483647, blockHash= 835bd577476f0cc0eea994b6f4c465b754781b4a264f52cc9f1b5c554fbbe073, isVerifyDB = false
2021-01-06 08:11:09 CheckTransaction nHeight=334680, isVerifyDB=false, isCheckWallet=false, txHash=1ea51bf9d0134bdaa7fed7e0b72b4c495fbad5f413471ced160c5b3889f16478
2021-01-06 08:11:09 CheckTransaction nHeight=334680, isVerifyDB=false, isCheckWallet=false, txHash=2e07f8e24b8c94aec5fe794a916cb813dec50c204ec882fc6635c5b16d4c908e
2021-01-06 08:11:09 CheckTransaction nHeight=334680, isVerifyDB=false, isCheckWallet=false, txHash=78aff886fe05dbbbcfd3084d4586190747b86a64e883369b6b85ce81160cf317
2021-01-06 08:11:09 CheckTransaction nHeight=334680, isVerifyDB=false, isCheckWallet=false, txHash=f9fffa18fe3951fb57f37241eb7cf96b38accd2856693a72219c148d4bfa524c
2021-01-06 08:11:09 AcceptBlock nHeight=334680
2021-01-06 08:11:09 CheckBlock() nHeight=334680, blockHash= 835bd577476f0cc0eea994b6f4c465b754781b4a264f52cc9f1b5c554fbbe073, isVerifyDB = false
2021-01-06 08:11:09 ERROR: AcceptBlock: , Stage 2 developer reward check failed (code 0)
2021-01-06 08:11:09 ERROR: ProcessNewBlock: AcceptBlock FAILED

and over place same error

2021-01-06 20:32:34 CheckBlock() nHeight=2147483647, blockHash= 5c597d620078ca611da2ce9e1a2ffb2a80fb937d467fec5a881e45b38140e588, isVerifyDB = false
2021-01-06 20:32:34 CheckTransaction nHeight=334822, isVerifyDB=false, isCheckWallet=false, txHash=a764f40fca2aaa05cfb79664b100163a65acc432a0941aeba74a42f33d613ee4
2021-01-06 20:32:34 CheckTransaction nHeight=334822, isVerifyDB=false, isCheckWallet=false, txHash=11676542820e8d2d60bb8f83b214c41fb550ca4d7ed850139fa4079f04954b31
2021-01-06 20:32:34 AcceptBlock nHeight=334822
2021-01-06 20:32:34 CheckBlock() nHeight=334822, blockHash= 5c597d620078ca611da2ce9e1a2ffb2a80fb937d467fec5a881e45b38140e588, isVerifyDB = false
2021-01-06 20:32:34 ERROR: AcceptBlock: , Stage 2 developer reward check failed (code 0)
2021-01-06 20:32:34 ERROR: ProcessNewBlock: AcceptBlock FAILED

How to resolve this problem?

**Comment by djm34** [comment]
currently solo mining is broken (due to the fact that founder reward has changed), I will try to fix it when I have some time

**Comment by sergeyzh71** [comment]
Dear djm34!
Solo mining is very importante part of mining. It"s great tradition from Satoshi Nakamoto! Please, try to find some time to correct ccminer. 
You write - "also considering I never received any dev fee with solo mining,...." It's imposible. I personally. to use your ccminer, mined Block #231343 in Jan 10, 2020 10:42:58 PM. 

https://explorer.zcoin.io/block/fc404a5d43c888d446b69eb90d8ab778546f411b2405ca2d854f9b1e6fbdef35

No Inputs (Newly Generated Coins)
13.93090428 FIRO (S)
a7h6tKgrnyNNoqeUo1t86e8pQ8CyBahWYp
0.5 FIRO (S)
aCAgTPgtYcA4EysU4UKC86EQd5cTtHtCcr
0.5 FIRO (S)
aHu897ivzmeFuLNB6956X6gyGeVNHUBRgD
0.5 FIRO (S)
aQ18FBVFtnueucZKeVg4srhmzbpAeb1KoN
1.5 FIRO (S)
a1HwTdCmQV3NspP2QqCGpehoFpi8NY4Zg3
0.5 FIRO (S)
a1kCCGddf5pMXSipLVD9hBG2MGGVNaJ15U
0.07 FIRO (S)
aChWVb8CpgajadpLmiwDZvZaKizQgHxfh5
7.5 FIRO (S)
aCieEDkd1bRy8FcvkcBXcdgBcb43cZ2vaf

I thought that 
0.07 FIRO (S)
aChWVb8CpgajadpLmiwDZvZaKizQgHxfh5
is your dev fee  with solo mining.
Is it wrong?
Sorry for my English. Thanks.

**Comment by djm34** [comment]
yes indeed that's my address... may-be I didn't pay attention... ok sorry my bad :) 
I'll try to fix it, but it may take some time, there are lots of change to put in place with the new masternode system.

**Comment by python8000** [comment]
Hey! I join the problem, the topic of solo mining is relevant, I ask you to modify the miner so that you can automatically deduct the commission when mining without a pool, that is, so that CCMiner independently divides the reward for the block. Thanks in advance for the prompt solution, and also thank you for the best miner for the mtp algorithm!

**Comment by djm34** [comment]
Hello, sorry for the delay, I am quite busy at the moment with other projects, I will try to implement the change, as soon as I have some times

**Comment by jackynanshan** [comment]
Hi djm34,

Does your ccminer support solo mining the dogecoin with dogecoind wallet? Below is my system error massage:
![image](https://user-images.githubusercontent.com/83000006/115837450-2f6b5d80-a44b-11eb-822f-e23be064ea86.png)

and below is my dogecoind running information:
![image](https://user-images.githubusercontent.com/83000006/115837699-73f6f900-a44b-11eb-875e-10e56069f09e.png)

If you can help me solve the problem I will appreciate you.

Thanks!

Jacky

**Comment by djm34** [comment]
the day doge moves to mtp, it will :D 
doge is using scrypt, you need asic to mine it, mining scrypt with gpu is a waste of electricity

**Comment by jackynanshan** [comment]
Hi djm34,

Yes, I know what the difference in MRT and scrypt now.

Thanks!

Jacky

发送自 Windows 10 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>应用

________________________________
发件人: djm34 ***@***.***>
发送时间: Friday, April 23, 2021 8:13:23 PM
收件人: firoorg/ccminer ***@***.***>
抄送: jackynanshan ***@***.***>; Comment ***@***.***>
主题: Re: [firoorg/ccminer] FIRO SOLO MINING ERROR (#68)


the day doge moves to mtp, it will :D
doge is using scrypt, you need asic to mine it.

―
